### ISSUE TITLE
Add Mastodon in social search

### ISSUE THREAD

**Issue by Popolon** [searx/searx#905]
Add the open source and decentralized [Mastodon](http://mastodon.social) network to the social search.

**Comment by dalf** [comment]
Just a note : it's not possible to do anonymous search, Mastodon requires to be log in to interact with it. 

It's not a deal breaker : some engine requires an API key).

**Comment by mangeurdenuage** [comment]
@Popolon @dalf 
Even if it was possible why support a project when the developer of the project is datamining and selling the data of it's users ?
https://mastodon.social/terms

Anyway it would be possible with the fedivers tho

**Comment by dalf** [comment]
@mangeurdenuage, that's precisely the purpose of searx : search information and keep the privacy.
Otherwise for the same reason searx shouldn't implements an engine for youtube or twitter for example.

The different is that mastodon is not so huge, so implementing an engine in searx will give more credit to this tool.

So should searx gives access to information what ever the TOS, or should a selection be made or should a warning be displayed as in F-Droid ? I don't know.

I've made a quick look around of the UI : I can't find a search in all the post ?

**Comment by mangeurdenuage** [comment]
@dalf 
--I've made a quick look around of the UI : I can't find a search in all the post ?--
It's been a while since I have used mastodon but I don't recall a search function in it to directly search.
I remember that the fediverse uses bots on mastodon to retrieve posts since the most active instances of the fediverse are blocked on the mastodon network.

--searx will give more credit to this tool.--
Do has you wish :)
But for myself I just don't support mastodon for various reasons, mostly it's because they are breaking the federate nature of the initial gnu social project and because of the possible embrace extend and extinguish that they may bring with [server wide ridiculous blocking list](https://github.com/WitchesTown/InstanceInternals) and the [on purpose protocol breaking](https://community.highlandarrow.com/notice/806880). 
Of course if mastodon would re-federate and stop all unethical actions it would be beneficial instead of generating piles of complaints that waste time.

--should a warning be displayed as in F-Droid--
The search via the searx would be more private than having to log in the mastodon instances of course but when they will be redirected to the mastodon network the TOS automatically applies (even if not read since the server is on American territory) thus meaning that the data and metadata will be exploited for diverse tracking and correlation.
In the TOS we can conclude that they use fingerprinting techniques.
See this part
" These cookies enable the site to recognize your browser and, if you have a registered account, associate it with your registered account."
They actually sell/share data with third parties
" This does not include trusted third parties who assist us in operating our site, conducting our business, or servicing you, so long as those parties agree to keep this information confidential. "
Note that even if those third parties agree to keep data confidential they contradict themselves with the line
" These third party sites have separate and independent privacy policies. We therefore have no responsibility or liability for the content and activities of these linked sites. "
Data and even metadata shouldn't not be shared or sold (legally speaking metadata is still recognized has data that isn't personal identifiable information which is of course not true)
for those curious see:
https://media.ccc.de/v/33c3-8414-corporate_surveillance_digital_tracking_big_data_privacy
and
https://media.libreplanet.org/mgoblin_media/media_entries/1529/144_7_gerwith.webm

A warning would help for awareness and also stand to the neutral state of the searx engine.
But it would demand either for human participation or if it is possible to use some lists that is already existing, for example in the µblock origin or umatrix lists.
For example in the "hpHosts’ Ad and tracking" servers list there's a lot of trackers that are already listed such has google-analytics.com
